src/StoreApi/Routes/Checkout.php

```diff
@@ -627,14 +567,31 @@ protected function process_customer( \WC_Order $order, \WP_REST_Request $request
 			try {
 				$create_account = Package::container()->get( CreateAccount::class );
 				$create_account->from_order_request( $request );
-				$order->set_customer_id( get_current_user_id() );
+				$this->order->set_customer_id( get_current_user_id() );
+				$this->order->save();
 			} catch ( Exception $error ) {
-				$this->handle_error( $error );
+				switch ( $error->getMessage() ) {
+					case 'registration-error-invalid-email':
+						throw new RouteException(
+							'registration-error-invalid-email',
+							__( 'Please provide a valid email address.', 'woo-gutenberg-products-block' ),
+							400
+						);
+					case 'registration-error-email-exists':
+						throw new RouteException(
+							'registration-error-email-exists',
+							apply_filters(
+								'woocommerce_registration_error_email_exists',
+								__( 'An account is already registered with your email address. Please log in.', 'woo-gutenberg-products-block' )
```

Comment:
I know this was moved here from `handle_error` method so was pre-existing to this pr, however we should probably implement WordPress coding standards for filters here. Might also be worth adding an inline dev note that any html will be escaped on the frontend when this is added to a notice in the blocks (because our notices system escapes html right?).

Reply:
I think we should remove this filter (I'm going to)—I can see it was added to mirror core, however, the filtered strings are different and the core filter allows for HTML whereas this would be more limited. Since this is an error from the API it does not need to be filterable. The client can change the wording based on the error code cc @haszari